File: 3420-find-occurrences-of-an-element-in-an-array/find-occurrences-of-an-element-in-an-array.java
class Solution {
    public int[] occurrencesOfElement(int[] nums, int[] queries, int x) {
        int n = nums.length;
        int j = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] == x) {
                nums[j] = i;
                j++;
            }
        }
        for (int i = 0; i < queries.length; i++) {
            if (queries[i] > j) queries[i] = -1;
            else queries[i] = nums[queries[i] - 1];
        }
        return queries;
    }
}

File: 3420-find-occurrences-of-an-element-in-an-array/find-occurrences-of-an-element-in-an-array.cpp
class Solution {
public:
    vector<int> occurrencesOfElement(vector<int>& nums, vector<int>& queries, int x) {
        map<int, int> mpp;
        int n=nums.size();
        int q=queries.size();
        vector<int>ans;
        int cnt = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] == x) {
                cnt++;
                mpp[cnt] = i;
            }
        }
        for (int i = 0; i < q; i++) {
            if (mpp.find(queries[i]) != mpp.end()) {
                ans.push_back(mpp[queries[i]]);
            } else {
                ans.push_back(-1);
            }
        }
        return ans;
    }
};